src/main/java/org/kainos/ea/resources/JobController.java

```diff
@@ -55,5 +59,22 @@ public Response getJobSpecById(@PathParam("id") int id) {
         }
     }
 
+    @PUT
+    @Path("/edit-job-role/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateJob(@PathParam("id") int id, JobRequest job) {
+        try {
+            jobService.updateJob(id, job);
+
+            return Response.ok().build();
+        } catch (InvalidJobException | JobDoesNotExistException e) {
+            System.err.println(e.getMessage());
+
+            return Response.status(Response.Status.BAD_REQUEST).entity(e.getMessage()).build();
```

Comment:
Handle these exceptions separately. When a job does not exist exception is thrown you should give back a 404 not found response.